backend/src/models/db/init_sql.js

```diff
@@ -0,0 +1,153 @@
+const { writeFileSync } = require('fs')
+const courses = require('./js/courses')
+const subjects = require('./js/subjects')
+const degreeData = require('../../../data/degrees')
+const faculties = require('../../../data/faculties')
+const { TABLE_NAMES } = require('../constants')
+
+const questionColumns = ['userID', 'code', 'title', 'body', 'pinned'].join(',')
+
+const ADMIN = 1
+const universityID = 1
+
+const users = [
+    {
+        id: ADMIN,
+        uid: '1233444',
+        displayName: 'Moderator',
+        email: 'admin@smartcourse.me',
+        degree: 'PhD',
+        gradYear: 2018
+    }
+]
+
+const SAMPLE_QUESTIONS = [
+    {
+        title: 'Is this course textbook necessary?',
+        body: 'I\'m wondering if it\'s worth buying/obtaining the course textbook. If not, are there good alternative resources that people might recommend?'
+    },
+    {
+        title: 'What year would you recommend taking this course?',
+        body: 'Is this course worth taking earlier or later in my degree? What are the advantages or disadvantages?'
+    },
+    {
+        title: 'Why did you take this course?',
+        body: 'I\'m wondering why people took this course as I don\'t know a lot about it.'
+    },
+    {
+        title: 'What are similar courses to this course?',
+        body: 'I\'m interested in this course\'s content area and I\'m looking for similar courses. What would people recommend, and why?'
+    },
+    {
+        title: 'How essential is it to attend lectures?',
+        body: 'Are the lectures for the course recorded? Are the worthwhile attending. What am I giving up if I can\'t attend them?'
+    }
+]
+
+// mega query begins here
+const data = `\
+BEGIN TRANSACTION;\n\
+${
+    // user
+    users.map(sqlUser).join('\n')
+}\n
+${
+    // uni
+    `INSERT INTO ${TABLE_NAMES.UNIVERSITY} (name) VALUES ("UNSW");`
+}\n
+${
+    // faculties
+    faculties.map(sqlFaculty).join('\n')
+}\n
+${
+    // degrees
+    degreeData.map(sqlDegree).join('\n')
+}\n
+${
+    // subjects
+    subjects.map(sqlSubject).join('\n')
+}\n
+${
+    // courses
+    courses.map(sqlCourse).join('\n')
+    // '\n'
+}\n
+${
+    // questions
+    courses.map(({ code }) =>
+        SAMPLE_QUESTIONS.map(sqlQuestion(code)).join('\n') + '\n').join('')
+}\
+\nCOMMIT;`
+
```

Comment:
Wow that is super clean

Reply:
Yep.

Reply:
Need to move the stuff in tables to be here as well. Unwind the last bit of that tables.js promise mess.